### ISSUE TITLE
Consider removing dependency on spring-boot-starter-web

### ISSUE THREAD

**Issue by viniciusccarvalho** [spring-cloud/spring-cloud-stream#1035]
Should we drop the need to pull web dependencies for stream apps? We had some comments on internal channels about issues when trying to use spring 5 reactive webflux, because we bring tomcat along.

I don't think we really need tomcat in first place, the apps should be working fine on console mode, unless someone needs actuator.

The downside here would be that the app-starters would probably need to bring the bom from boot.

I was able to create a version of the project without any boot-web (we still need spring-web for converters) so I don't think this is a major work.

Open to discuss with others

**Comment by olegz** [comment]
@viniciusccarvalho I would look at it from the standpoint of what will be lost if we drop the web as a dependency. For example all those REST endpoints for health and monitoring will no longer be available until one explicitly adds web starter. gone. Is this something we're ok with? The flip argument could be "if you need those then add web starter", so in reality it is a question of sensible default.

**Comment by viniciusccarvalho** [comment]
That's the reason for the actuator, if someone needs any of the management endpoints, they should bring it along. 

Marius and I discussed this a while ago, and we could not find a strong reason to keep the web dependency as a strong requirement.

Note that the apps are still going to depend on it, so any of the OOB starters should have the actuator endpoints along with tomcat.

But for people using SCSt standalone, they may not be interested in using tomcat, or even be a web app at all.

The problem we got, is that it seems that boot disables WebFlux if it detects tomcat on the classpath.
